### PR TITLE
 oidc: Use the correct types to compare the status codes in the oidc example

### PR DESCRIPTION
**Pull request by poljar**
Something went wrong while #3305 was merged. This fixes the issue introduced by the github rebase.

### CODE REVIEW

**Comment by zecakeh** [comment]
I believe this should not be necessary if #3362 is merged.

**Comment by zecakeh** [comment]
Actually, we could fully qualify `StatusCode` in this case with `matrix_sdk::reqwest::StatusCode`.

**Comment by poljar** [comment]
> I believe this should not be necessary if #3362 is merged.

Hah yeah, though I think it makes sense to fix the bodged merge of #3305 with the smallest possible patch.

**Comment by poljar** [comment]
> Actually, we could fully qualify `StatusCode` in this case with `matrix_sdk::reqwest::StatusCode`.

You mean post merge or in this PR?

**Comment by zecakeh** [comment]
> You mean post merge or in this PR?

In this PR, that should work at any time.

 Actually the best option would probably be to not depend directly on http, but use `axum::http` types for interactions with axum types, and `matrix::reqwest` types for interacting with the SDK's types.

**Review by bnjbvr — APPROVED** [review]

**Comment by poljar** [comment]
> > You mean post merge or in this PR?
> 
> In this PR, that should work at any time.
> 
> Actually the best option would probably be to not depend directly on http, but use `axum::http` types for interactions with axum types, and `matrix::reqwest` types for interacting with the SDK's types.

Right: https://github.com/matrix-org/matrix-rust-sdk/pull/3363/commits/344c77bcec6c16b2e48c62fbb11d9a30db20be7d

File: examples/oidc_cli/src/main.rs [hunk 1]
```diff
@@ -23,9 +23,12 @@ use std::{
 };
 
 use anyhow::{anyhow, bail};
-use axum::{response::IntoResponse, routing::any_service};
+use axum::{
+    http::{Method, StatusCode},
```
Comment (zecakeh) [inline examples/oidc_cli/src/main.rs:27]: The http dependency needs to be removed from the `Cargo.toml`.

**Pull request merged**